samples/dotnet/KernelBuilder/Program.cs

```diff
@@ -122,6 +122,10 @@
 
 var kernel10 = Kernel.Builder.WithRetryHandlerFactory(new RetryThreeTimesFactory()).Build();
 
+// ==========================================================================================================
+// Kernel instance should be disposed to release OS resources it may hold, when it's not required anymore.
```

Comment:
what happens when disposing 2 or more kernels that share skill objects?

Reply:
They won't be disposed because Colectivismo is not disposable today (there's another PR that will dispose skills and we need to hold it until proper solution is found). However, semantic memory is disposable, and it'll cause a problem if it's used by more than one Kernel and one of them is disposed. Though, I don't know if one memory instance is supposed to be used by more than one Kernel, but Kernel constructor allows this.

I propose to change the way Kernel deals with its dependencies that are disposable and may be shared between many kernel instances - don't create the dependencies (memory, skillscollection, etc) in Kernel/KernelBuilder, delegate their creation to a hosting app so that it has to supply/inject them through Kernel constructor and KernelBuilder.Build method. 

This simple change will:
- simplify the Kernel - no need to be disposable anymore because it won't own the resources, it just uses the ones provided/injected and as such it is not supposed to dispose them.
- allow the hosting app to fully control lifetime of the sharable and disposable resources and mix and match them with as many Kernel instances as it needs.

Reply:
I'm wondering, do we have a use-case when 2 or more kernel instances can be helpful for the app? What if kernel will be used in applications as a singleton?